x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/field_parameters/inference_id_selects.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiCallOut, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import React, { useEffect, useState } from 'react';
+
+import { InferenceAPIConfigResponse } from '@kbn/ml-plugin/common/types/trained_models';
```

Comment:
Move this type definition from the ML plugin to `x-pack/packages/ml/trained_model_utils`, then import it from there. That should avoid the cyclical dependency you're seeing now. We also should not be importing from deep inside a different plugin anyway.